libbeat/common/transport/tlscommon/tls_config.go

```diff
@@ -86,68 +95,174 @@ func (c *TLSConfig) ToConfig() *tls.Config {
 
 	// When we are using the CAsha256 pin to validate the CA used to validate the chain,
 	// or when we are using 'certificate' TLS verification mode, we add a custom callback
-	verifyPeerCertFn := makeVerifyPeerCertificate(c)
+	verifyConnectionFn := makeVerifyConnection(c)
 
-	insecure := c.Verification != VerifyFull
+	insecure := c.Verification != VerifyStrict
 	if c.Verification == VerifyNone {
 		logp.NewLogger("tls").Warn("SSL/TLS verifications disabled.")
 	}
-
 	return &tls.Config{
-		MinVersion:            minVersion,
-		MaxVersion:            maxVersion,
-		Certificates:          c.Certificates,
-		RootCAs:               c.RootCAs,
-		ClientCAs:             c.ClientCAs,
-		InsecureSkipVerify:    insecure,
-		CipherSuites:          c.CipherSuites,
-		CurvePreferences:      c.CurvePreferences,
-		Renegotiation:         c.Renegotiation,
-		ClientAuth:            c.ClientAuth,
-		VerifyPeerCertificate: verifyPeerCertFn,
-		Time:                  c.time,
+		MinVersion:         minVersion,
+		MaxVersion:         maxVersion,
+		Certificates:       c.Certificates,
+		RootCAs:            c.RootCAs,
+		ClientCAs:          c.ClientCAs,
+		InsecureSkipVerify: insecure,
+		CipherSuites:       c.CipherSuites,
+		CurvePreferences:   c.CurvePreferences,
+		Renegotiation:      c.Renegotiation,
+		ClientAuth:         c.ClientAuth,
+		Time:               c.time,
+		VerifyConnection:   verifyConnectionFn,
 	}
 }
 
 // BuildModuleConfig takes the TLSConfig and transform it into a `tls.Config`.
-func (c *TLSConfig) BuildModuleConfig(host string) *tls.Config {
+func (c *TLSConfig) BuildModuleClientConfig(host string) *tls.Config {
 	if c == nil {
 		// use default TLS settings, if config is empty.
-		return &tls.Config{ServerName: host}
+		return &tls.Config{
+			ServerName:         host,
+			InsecureSkipVerify: true,
+			VerifyConnection: makeVerifyConnection(&TLSConfig{
+				Verification: VerifyFull,
+			}),
+		}
 	}
 
 	config := c.ToConfig()
 	config.ServerName = host
+	config.VerifyConnection = makeVerifyConnection(c)
```

Comment:
`VerifyConnection` seems to be set by `ToConfig` now. We still need this line?

Do we need to set `config.ServerName` given that we configure TLS for the client use?